pkg/build/builder/cmd/builder.go

```diff
@@ -107,19 +110,18 @@ func (c *builderConfig) setupGitEnvironment() ([]string, error) {
 		//   it accepts
 		sourceURL, err := git.ParseRepository(gitSource.URI)
 		if err != nil {
-			return nil, fmt.Errorf("cannot parse build URL: %s", gitSource.URI)
+			return "", nil, fmt.Errorf("cannot parse build URL: %s", gitSource.URI)
 		}
 		scmAuths := scmauth.GitAuths(sourceURL)
 
 		// TODO: remove when not necessary to fix up the secret dir permission
-		sourceSecretDir, err := fixSecretPermissions(c.sourceSecretDir)
-		if err != nil {
-			return nil, fmt.Errorf("cannot fix source secret permissions: %v", err)
+		sourceSecretDir, errSecret = fixSecretPermissions(c.sourceSecretDir)
+		if errSecret != nil {
+			return sourceSecretDir, nil, fmt.Errorf("cannot fix source secret permissions: %v", errSecret)
```

Comment:
If an error occurred you should delete the secret directory and then return the error. Otherwise, the secret directory will not get removed (the caller function simply returns the error -- line 151)